### PR TITLE
Warehouse Asset Gem

### PR DESCRIPTION
**Pull request by michalpelka**
- [x] Accept the name of the gem
- [x] List of contributors
- [x] Add icon
- [x] Sample scene prefab
- [x] Adjust the structure in "Assets" directory - will be done in separate PR.

### CODE REVIEW

**Comment by pawelbudziszewski** [comment]
Warehouse works exactly as it should. Good job!

Some comments from my side:
1. Since we already have a `WarehouseSample` gem, the name is a bit confusing. 
2. I'd consider embedding a camera in the main prefab. Some visual/post-processing effects are defined using camera components, IMHO it would be nice to have a "reference camera" in the project.
3. It would be nice to do some cleaning:
 - There should be only the main prefab directly in the `Assets/Prefabs` folder, others should be moved to sub-folders (this is relatively easy to do)
 - All materials/textures should be moved from the top level of `Assets/assets` to sub-folders (this may require updating them in the main prefab)
4. Maybe we should consider adding a README to this gem? We could describe the structure of prefabs and basic instructions for creating  and adjusting halls

**Review by pawelbudziszewski — APPROVED** [review]

**Comment by hultonha** [comment]
question: It might be possible to do this after but I wanted to check if it might be worth some of these robotics related Gems in a robotics subfolder in the O3DE-Extras Gem folder? It's not a huge deal but then just might help with the logical groupings of related Gems (tagging @lumberyard-employee-dm for their thoughts). Thanks!

**Comment by adamdbrw** [comment]
This PR contributes to #182

File: Gems/WarehouseAssets/Assets/Prefabs/Warehouse_structural/Components/Ceeling_Lamp.prefab [hunk 1]
```diff
@@ -0,0 +1,374 @@
+{
+    "ContainerEntity": {
+        "Id": "ContainerEntity",
+        "Name": "Ceeling_Lamp",
```
Comment (hultonha) [inline Gems/WarehouseAssets/Assets/Prefabs/Warehouse_structural/Components/Ceeling_Lamp.prefab:4]: nit: This is super minor but I think this should be `ceiling` lamp.

File: Gems/WarehouseAssets/Code/Include/WarehouseAssets/WarehouseAssetsBus.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Interface/Interface.h>
+
+namespace WarehouseAssets
+{
+    class WarehouseAssetsRequests
```
Comment (hultonha) [inline Gems/WarehouseAssets/Code/Include/WarehouseAssets/WarehouseAssetsBus.h:9]: question: As this is an asset only Gem (or at least appears to be) could we delete all this autogenerated code? Is it actually needed for anything? Again @lumberyard-employee-dm might be able to weigh in on this. Thanks!

**Review by hultonha — COMMENTED** [review]
Makes sense to me but I would like to know if it's possible to actually delete some (or all?) of the generated code as I'm not sure if all of it is needed for this asset only Gem. Once that's answered I can approve 👍 Thanks!

File: Gems/WarehouseAssets/Code/Include/WarehouseAssets/WarehouseAssetsBus.h (hunk 2 above)
Comment (michalpelka) [reply Gems/WarehouseAssets/Code/Include/WarehouseAssets/WarehouseAssetsBus.h]: Changed to asset Gem and removed code.

File: Gems/WarehouseAssets/Assets/Prefabs/Warehouse_structural/Components/Ceeling_Lamp.prefab (hunk 1 above)
Comment (michalpelka) [reply Gems/WarehouseAssets/Assets/Prefabs/Warehouse_structural/Components/Ceeling_Lamp.prefab]: Fixed typo.

**Comment by lumberyard-employee-dm** [comment]
> question: It might be possible to do this after but I wanted to check if it might be worth some of these robotics related Gems in a robotics subfolder in the O3DE-Extras Gem folder? It's not a huge deal but then just might help with the logical groupings of related Gems (tagging @lumberyard-employee-dm for their thoughts). Thanks!

Yes the robotics gems can be moved to a "Robotics" subfolder if that is desired.
The enable-gem and register commands support specifying a folder for gems where recursively gems are registered within.
The `o3de.py register` command uses the [all-gems-path](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/register.py#L915-L916) option (or `-agp`) for registering a set of gems under a folder, while the `o3de.py enable-gem` command uses the **differently named** [all-gem-paths](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/enable_gem.py#L226-L227) option (or `-agp`) for explicitly activating gems within a folder.

(Side Note: if using the long options note the difference between all-gems-path, where "gems" have an 's' vs all-gem-paths, where "paths" has an 's').

**Comment by hultonha** [comment]
Thanks for the info @lumberyard-employee-dm! 👍

File: Gems/WarehouseAssets/Code/Include/WarehouseAssets/WarehouseAssetsBus.h (hunk 2 above)
Comment (hultonha) [reply Gems/WarehouseAssets/Code/Include/WarehouseAssets/WarehouseAssetsBus.h]: Awesome thank you!

File: Gems/WarehouseAssets/Assets/Prefabs/Warehouse_structural/Components/Ceeling_Lamp.prefab (hunk 1 above)
Comment (hultonha) [reply Gems/WarehouseAssets/Assets/Prefabs/Warehouse_structural/Components/Ceeling_Lamp.prefab]: Thanks!

**Review by hultonha — APPROVED** [review]
Looks great thanks!

**Comment by michalpelka** [comment]
Failed build seems to be unrelated to changes:
```
[2023-02-21T10:37:49.606Z] D:/workspace/o3de-extras/Gems/XR/Code/Source/XRSystemComponent.cpp(48): error C2259:             m_xrSystem = aznew System(); [D:\workspace\o3de\build\windows\External\XR-7f8e4980\Code\XR.Static.vcxproj]

[2023-02-21T10:37:49.866Z] D:\workspace\o3de-extras\Gems\XR\Code\Include\XR/XRSystem.h(27): message : see declaration of 'XR::System' [D:\workspace\o3de\build\windows\External\XR-7f8e4980\Code\XR.Static.vcxproj]

[2023-02-21T10:37:49.875Z] D:\workspace\o3de-extras\Gems\XR\Code\Include\XR/XRSystem.h(27): message :     class System [D:\workspace\o3de\build\windows\External\XR-7f8e4980\Code\XR.Static.vcxproj]

[2023-02-21T10:37:50.136Z] D:/workspace/o3de-extras/Gems/XR/Code/Source/XRSystemComponent.cpp(48): message : due to following members: [D:\workspace\o3de\build\windows\External\XR-
```

**Comment by hultonha** [comment]
> Failed build seems to be unrelated to changes in this PR.
> 
> ```
> [2023-02-21T10:37:49.606Z] D:/workspace/o3de-extras/Gems/XR/Code/Source/XRSystemComponent.cpp(48): error C2259:             m_xrSystem = aznew System(); [D:\workspace\o3de\build\windows\External\XR-7f8e4980\Code\XR.Static.vcxproj]
> 
> [2023-02-21T10:37:49.866Z] D:\workspace\o3de-extras\Gems\XR\Code\Include\XR/XRSystem.h(27): message : see declaration of 'XR::System' [D:\workspace\o3de\build\windows\External\XR-7f8e4980\Code\XR.Static.vcxproj]
> 
> [2023-02-21T10:37:49.875Z] D:\workspace\o3de-extras\Gems\XR\Code\Include\XR/XRSystem.h(27): message :     class System [D:\workspace\o3de\build\windows\External\XR-7f8e4980\Code\XR.Static.vcxproj]
> 
> [2023-02-21T10:37:50.136Z] D:/workspace/o3de-extras/Gems/XR/Code/Source/XRSystemComponent.cpp(48): message : due to following members: [D:\workspace\o3de\build\windows\External\XR-
> ```

Hmm that does seem very odd (especially as this change is asset only right?) Does this branch need updating from development? There does seem to be something a bit strange with the build machines for o3de-extras potentially. I can reach out to infra for support if we can't seem to root cause this.

**Comment by michalpelka** [comment]
Rebased.

**Comment by hultonha** [comment]
> Rebased on top of development

Okay thanks @michalpelka, I'll try another build now

**Comment by hultonha** [comment]
https://jenkins.build.o3de.org/view/Projects/job/o3de-extras/view/change-requests/job/PR-174/2/ - if this fails again we might need to investigate (as far as I understand this PR has no code changes so I'm not sure why the previous error was suddenly being reported).

**Comment by shawstar** [comment]
I think the main readme should remind customers to install and pull lfs as this is not just a code gem.

**Comment by shawstar** [comment]
Always nice to show illustrations to customers in your gem Readme. Customers can also add gems with the Project Manager to the Gem Catalog. 
image

**Comment by shawstar** [comment]
Totally being picky here. But if I place the Warehouse Prefab at world 0,0,0 it will be slightly lower in elevation than 0. Would be better if the ground level of the Warehouse is at 0, 
image

**Comment by shawstar** [comment]
I see opportunities when it comes to the look development of the interior lighting. 
image

I really liked the demo warehouse video, that had a very good look. Would be great if this gem sample was to that visual level.

**Comment by shawstar** [comment]
I really like the detail of the next prefab, the warehouse scene. Even with out adding some camera effects, it features are more detailed, and to a more realistic scale. The 1st warehouse scene, had opportunities in this area. 
image

**Comment by shawstar** [comment]
Might be cool to add some postfx and some reflection probes. I just added abit and enhanced the look. 
image

**Comment by shawstar** [comment]
Yeah. Looks good. Technically I had no issues. There is just one thing that customers need to know to make sure to use  install and pull lfs when using this gem repo, or extras.

**Pull request merged**